Source/Particles/Collision/BackgroundStopping/BackgroundStopping.cpp

```diff
@@ -84,6 +84,9 @@ BackgroundStopping::doCollisions (amrex::Real cur_time, amrex::Real dt, MultiPar
     amrex::Real species_mass = species.getMass();
     amrex::Real species_charge = species.getCharge();
 
+    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(species_mass > 0., "Error: With background stopping, the species mass must be > 0");
+    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(species_charge != 0., "Error: With background stopping, the species charge must be nonzero");
```

Comment:
```C++
#include <cmath>
#include <limits>
```
and
```suggestion
    amrex::Real const species_mass = species.getMass();
    amrex::Real const species_charge = species.getCharge();

    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(species_mass > 0._rt,
        "Error: With background stopping, the species mass must be > 0");
    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(std::fabs(species_charge) > std::numeric_limits<amrex::Real>::epsilon(),
        "Error: With background stopping, the species charge must be nonzero");
```

note: the 2nd one should better be a comparison to an epsilon, since exact `==`/`!=` is flaky for floating point numbers.

Reply:
I agree that in general, checking for floating point equality is problematic. However, checking for `x != 0.` should be robust since `0` is an exact floating point value.

Reply:
Yes, that's true but -0.0 exits as well and maybe the value comes from a calc/parser thingy and is only close to zero.

Reply:
In C++, the expression `-0.0 == 0.0` is true, so that is not an issue. Also, the value of `epsilon` is 2.22045e-16 so this check would fail since `fabs(species_charge)` would likely be 1.6e-19.